torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -449,10 +449,11 @@ class ShapePropagator {
       }
     }
     node->output()->setType(
-        DimensionedTensorType::create(default_type, default_device, dims));
+        ProfiledTensorType::create(default_type, default_device, dims, true));
```

Comment:
nitpick: `/*requires_grad=*/ true`

Reply:
Also, shouldn't all those places have requires_grad set to unknown?

Reply:
Either unknown or true will end up doing the same thing. This was to preserve what dimensioned tensor did, which was always `true` by default. I can change it though.

Reply:
Well unless we want to use the same overapproximation behavior as in DimensionedTensorType, then we should use unknown, because that's the right thing to o in this case. I thought the idea is for profiled type to be a bit of a cleanup, and this seems like a better idea to me.